### PR TITLE
Separate some types between taskManager and taskStore

### PR DESCRIPTION
**Pull request by anish531213**
Separate some types between taskManager and taskStore that have timer type as int64

### CODE REVIEW

File: common/persistence/persistenceInterface.go [hunk 1]
```diff
@@ -46,8 +46,30 @@ type (
 		GetShard(ctx context.Context, request *InternalGetShardRequest) (*InternalGetShardResponse, error)
 		UpdateShard(ctx context.Context, request *InternalUpdateShardRequest) error
 	}
+
 	// TaskStore is a lower level of TaskManager
-	TaskStore = TaskManager
+	TaskStore interface {
+		Closeable
+		GetName() string
+		LeaseTaskList(ctx context.Context, request *LeaseTaskListRequest) (*LeaseTaskListResponse, error)
```
Comment (andrewjdawson2016) [inline common/persistence/persistenceInterface.go:54]: `LeaseTaskListRequest` takes a TaskListKind which is an int. That in represents a value of a thrift enum. Which is a serialization detail. Therefore `LeaseTaskListRequest` should be separated.

Comment (andrewjdawson2016) [reply common/persistence/persistenceInterface.go:54]: The same is true of `LeaseTaskListResponse`

**Review by andrewjdawson2016 — CHANGES_REQUESTED** [review]

File: common/persistence/persistenceInterface.go (hunk 1 above)
Comment (andrewjdawson2016) [reply common/persistence/persistenceInterface.go:54]: There are other cases in which this is true as well please take a another pass over those request and response structs. I think many of them have taskListKind. We will want to represent task list kind as an application level enum eventually.

Comment (anish531213) [reply common/persistence/persistenceInterface.go:54]: The taskListKind is an enum in dataInterface. So, I didn't understand how it is related to serialization here.

**Review by andrewjdawson2016 — APPROVED** [review]

**Pull request merged**